### DISCUSSION TITLE
Reading Files During Runtime in an Onefile-Application

### DISCUSSION THREAD

**Discussion by l-grasl** [pyinstaller/pyinstaller#7709] [Help]
Hi there,
I am trying to build an Executable (on Windows11) from a Python application which processes data from an xlsx / csv - input. 
Importing the files and libraries that come with the application (icon etc.) is working - but the files I need to read during the application (the input data the application is supposed to process) are not being read.
I think it's a problem with the pathvariables, but can't find a way to make it work. 
Can someone help with this?
Thanks :)

**Comment by bwoodsend** [comment]
Is this xlsx/csv supposed to be part of the application or passed in as a CLI argument? Docs for the former (adding data files and finding them at runtime are [here](https://pyinstaller.org/en/stable/runtime-information.html?highlight=__file__#using-file)) although I recommend temporarily switching to onedir mode so you can see if/where the file is collected to. The latter should work out the box.

**Reply by l-grasl** [reply]
Thank's for your reply. The file is passed in as an argument and is not part of the application itself. In onedir-Mode everything is working fine but for distributing I want to convert it to an onefile app.

**Reply by bwoodsend** [reply]
I'd be very surprised if the reason isn't that you're doing something invalid getting this path then. The simplest example I can come up with works perfectly. 

```python
# test.py
import sys
import pandas
print(pandas.read_csv(sys.argv[1]))
```

```bash
> pyinstaller --onefile test.py
> ./dist/test test.csv
   a  b  c
0  1  2  3
1  4  5  6
```
I can't tell you anything more specific without seeing your code.
